v-next/core/src/internal/hook-manager.ts

```diff
@@ -288,30 +293,31 @@ export class HookManagerImplementation implements HookManager {
     path: string,
   ): Promise<Partial<HardhatHooks[HookCategoryNameT]>> {
     if (!path.startsWith("file://")) {
-      throw new Error(
-        `Plugin ${pluginId} hook factory for ${hookCategoryName} is not a valid file:// URL: ${path}`,
+      throw new HardhatError(
+        HardhatError.ERRORS.HOOKS.INVALID_HOOK_FACTORY_PATH,
```

Comment:
For my curiosity: why this is the only error coming from HardhatError.ERRORS?

Reply:
That is the only error left in `HookManager`, the other errors have been converted to assertions. I limited the conversion to `HardhatError` to only those in the code under test.